feat: enhance registry functionality with new search API integration and UI improvements

siamak opened the pull request:
- Added a redirect for the "/registry" route to the external registry URL.
- Introduced a new mutedForeground color palette in Tailwind configuration.
- Updated the registry page to conditionally apply filters based on the new search API response.
- Improved the layout of the Codemod list and registry section cards for better responsiveness.
- Enhanced the mobile and desktop filter components to utilize the new filter types.
- Updated the data fetching logic to accommodate the new API structure and added error handling.
- Refactored various components to streamline code and improve readability.

siamak committed:
feat: enhance registry functionality with new search API integration and UI improvements

- Added a redirect for the "/registry" route to the external registry URL.
- Introduced a new mutedForeground color palette in Tailwind configuration.
- Updated the registry page to conditionally apply filters based on the new search API response.
- Improved the layout of the Codemod list and registry section cards for better responsiveness.
- Enhanced the mobile and desktop filter components to utilize the new filter types.
- Updated the data fetching logic to accommodate the new API structure and added error handling.
- Refactored various components to streamline code and improve readability.

siamak committed:
refactor: update registry routing and remove deprecated components

- Added a new redirect for the "/registry/:path*" route to point to the external Codemod registry.
- Removed the obsolete RegistryIndex and CodemodRoute components to streamline the codebase.
- Updated helper functions to reflect the new external registry URL.
- Maintained backward compatibility in the loadRegistryIndex function while returning null for callers.

siamak committed:
feat: add support for legacy registry endpoint and improve registry routing

- Introduced a new environment variable for the old Codemod automations list endpoint.
- Updated the loadCodemod function to conditionally use the old endpoint based on a new parameter.
- Added a new route for handling individual codemod requests, redirecting to the appropriate registry URL.
- Enhanced the RegistrySectionCard component to display organization and owner avatars correctly.

siamak committed:
refactor: clean up registry components and improve code readability

- Removed unnecessary blank lines and improved formatting in various components for better readability.
- Added a new route for handling registry searches with query parameters for frameworks, categories, and authors.
- Introduced a new TAGS structure for managing framework tags and their aliases.
- Updated the loadCodemod function to enhance error handling and streamline API calls.
- Refactored the DesktopFilters and RegistryCard components to improve layout and maintainability.

mohebifar commented on apps/frontend/app/(website)/registry/route.tsx:
```diff
@@ -0,0 +1,379 @@
+  return framework ? parseAliases(framework.aliases) : [];
+}
+
+/**
+ * Parses strings like:
+ *  "{next.js,nextjs,next}"
+ *  "{\"best practices\"}"
+ * into: ["next.js","nextjs","next"] or ["best practices"]
+ */
+function parseAliases(raw: string): string[] {
```
I'd recommend we simply update the base JSON object's aliases prop to be an array of strings instead of parsing it in runtime

siamak committed:
refactor: streamline TAGS structure and enhance alias lookup functionality

- Replaced the previous TAGS structure with a more organized Record-based format for frameworks and use case categories.
- Implemented a new alias lookup function to improve the handling of framework and category aliases in search queries.
- Removed deprecated functions related to alias parsing and normalization to simplify the codebase.
- Updated the GET handler to utilize the new alias lookup logic for better search results.

siamak committed:
refactor: improve TAGS structure for better clarity and consistency

- Enhanced the TAGS definition by adjusting formatting for improved readability.
- Corrected the key for "best practices" in the useCaseCategory to align with naming conventions.

siamak committed:
refactor: enhance GET handler for improved redirection logic

- Updated the GET handler to redirect to the correct registry URL based on the codemod parameter.
- Improved error handling by ensuring a fallback redirect to the main registry URL in case of errors.
- Cleaned up formatting for better readability and consistency in the code structure.

siamak committed:
refactor: enhance GET handler with consistent parameter formatting

- Updated the GET handler to ensure consistent formatting of parameters.
- Improved readability by adding missing commas in object literals.
- Maintained existing functionality while enhancing code clarity.
